app/src/lib/stores/app-store.ts

```diff
@@ -5246,22 +5249,18 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
       await this._fetchRemote(repository, remote, FetchType.UserInitiatedTask)
 
-      const gitStore = this.gitStoreCache.get(repository)
-
       const localBranchName = `pr/${pullRequest.pullRequestNumber}`
-      const doesBranchExist =
-        gitStore.allBranches.find(branch => branch.name === localBranchName) !=
-        null
+      const existingBranch = this.getLocalBranch(repository, localBranchName)
 
-      if (!doesBranchExist) {
+      if (existingBranch === null) {
         await this._createBranch(
           repository,
           localBranchName,
           `${remoteName}/${head.ref}`
         )
+      } else {
```

Comment:
This `else` is new. Just wanted to double check that the behavior is intended... my understanding is that previously if the branch didn't exist we would create it and then check it out. Now, with this `else` we only create the branch and _do not_ check it out? @niik

Reply:
Oh yeah, I should have documented this, sorry about that. If you look in `_createBranch` we already do a checkout as part of creating the branch

https://github.com/desktop/desktop/blob/30305ab02a1efef6404f940e3443a5805c10c41c/app/src/lib/stores/app-store.ts#L3000-L3006

In other words, the previous `_checkoutBranch` call was redundant.